packages/compass-saved-aggregations-queries/src/util/format-date.tsx

```diff
@@ -0,0 +1,50 @@
+const SECOND = 1000;
+const MINUTE = SECOND * 60;
+const HOUR = MINUTE * 60;
+const DAY = HOUR * 24;
+const WEEK = DAY * 7;
+const MONTH = WEEK * 4;
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-expect-error this does exist, just not in TS types
+const relativeDateFormatter = new Intl.RelativeTimeFormat('en', {
+  numeric: 'auto',
+});
+
+const absoluteDateFormatter = new Intl.DateTimeFormat('en', {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-expect-error this does exist, just not in TS types
+  dateStyle: 'long',
+});
+
+const formatScale = {
+  second: SECOND,
+  minute: MINUTE,
+  hour: HOUR,
+  day: DAY,
+  week: WEEK,
+};
+
+export function formatDate(date: number): string {
+  const timeDiff = Date.now() - date;
```

Comment:
Not sure how useful this is, but we use `moment` in other parts of Compass, it would be probably good to get some consistency.

Reply:
I know that we use it, but I would prefer not to use [in maintenance](https://github.com/moment/moment/#project-status) but discouraged to be used library when better / lighter built-in browser alternatives exist. I'm happy to switch if you feel like consistency is more important here

Reply:
Maybe we should do it the other way around (in a separate PR). Let me quickly check how it is used, I'll create a ticket.

Reply:
Do we know their browser support policy? `RelativeTimeFormat` is the least supported one, but even this one is available in chrome for ~4 years now and in ff for ~3 years. Polyfills are also available for all this stuff, yep